src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -163,24 +163,44 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
             manager, "GET",
             QUrl(QString("%1/virtual-machines/%2").arg(base_url.toString()).arg(QString::fromStdString(query.name))));
 
-        auto id = instance_info["metadata"].toObject()["config"].toObject()["volatile.base_image"].toString();
+        auto config = instance_info["metadata"].toObject()["config"].toObject();
+
+        if (config.contains("image.original_hash"))
+        {
+            VMImage source_image;
+
+            source_image.id = config["image.original_hash"].toString().toStdString();
+            source_image.original_release = config["image.description"].toString().toStdString();
+            source_image.release_date = config["image.version"].toString().toStdString();
+
+            return source_image;
+        }
+
+        auto id = config["volatile.base_image"].toString();
 
         for (const auto& image_host : image_hosts)
         {
-            auto info = image_host->info_for_full_hash(id.toStdString());
+            try
+            {
+                auto info = image_host->info_for_full_hash(id.toStdString());
 
-            VMImage source_image;
+                VMImage source_image;
 
-            source_image.id = id.toStdString();
-            source_image.original_release = info.release_title.toStdString();
-            source_image.release_date = info.version.toStdString();
+                source_image.id = id.toStdString();
+                source_image.original_release = info.release_title.toStdString();
+                source_image.release_date = info.version.toStdString();
+
+                for (const auto& alias : info.aliases)
+                {
+                    source_image.aliases.push_back(alias.toStdString());
+                }
 
-            for (const auto& alias : info.aliases)
+                return source_image;
+            }
+            catch (const std::exception&)
             {
-                source_image.aliases.push_back(alias.toStdString());
+                // Ignore
             }
```

Comment:
Could we be stricter here?

Reply:
Well, `info_for_full_hash()` will throw a `std::runtime_error` when it doesn't find a matching hash, but there are multiple image hosts, so we need to ignore that exception so we can move on in the loop.  So the only better granularity we can have here is to catch the `runtime_error` explicitly, but that is quite generic in and of itself.

Reply:
Add a TODO, please.

Reply:
A TODO for what?  Change the exception type that `info_for_full_hash()` throws?  I don't think changing this to catch `std::runtime_error` is the right thing to do.

Reply:
Yes, for refactoring so that `info_for_full_hash` throws specific exceptions.

Reply:
Ok, I added a TODO here and in the place in `ubuntu_image_host.cpp` where it really needs to be implemented.